### PR TITLE
Add power governor integration test to new pattern

### PR DESCRIPTION
**Pull request by cmcantalupo**
Separate out power governor test from geopm_test_integration

### CODE REVIEW

File: integration/test/test_ompt.py [hunk 1]
```diff
@@ -53,7 +53,7 @@ def setUpClass(cls):
                          '.' + cls.__name__ + ') ...')
         test_name = 'test_ompt'
         cls._report_path = '{}.report'.format(test_name)
-        cls._agent_conf_path = 'test_' + test_name + '-agent-config.json'
+        cls._agent_conf_path = test_name + '-agent-config.json'
```
Comment (bgeltz) [inline integration/test/test_ompt.py:56]: Was this change to the OMPT code intentionally included?

File: integration/test/test_power_governor.py [hunk 2]
```diff
@@ -34,73 +34,68 @@
...
+        cls._trace_path = '{}.trace'.format(test_name)
+        cls._agent_conf_path = test_name + '-agent-config.json'
+        # Clear out exception record for python 2 support
+        geopmpy.error.exc_clear()
+        # Set the job size parameters
+        cls._num_node = 4
         num_rank = 16
         loop_count = 500
-        app_conf = geopmpy.io.BenchConf(name + '_app.config')
-        self._tmp_files.append(app_conf.get_path())
```
Comment (bgeltz) [inline integration/test/test_power_governor.py]: I see that tmp_files tracking has been removed.  Does this mean these tests will be littering files all over the place?

File: integration/test/test_power_governor.py [hunk 3]
```diff
@@ -114,11 +109,8 @@ def test_power_consumption(self):
             pkg_energy_cols = [s for s in power_data.keys() if 'ENERGY_PACKAGE' in s]
             dram_energy_cols = [s for s in power_data.keys() if 'ENERGY_DRAM' in s]
             power_data['SOCKET_POWER'] = power_data[pkg_energy_cols].sum(axis=1) / power_data['ELAPSED_TIME']
-            power_data['DRAM_POWER'] = power_data[dram_energy_cols].sum(axis=1) / power_data['ELAPSED_TIME']
-            power_data['COMBINED_POWER'] = power_data['SOCKET_POWER'] + power_data['DRAM_POWER']
 
             pandas.set_option('display.width', 100)
-            launcher.write_log(name, 'Power stats from {} :\n{}'.format(nn, power_data.describe()))
```
Comment (bgeltz) [inline integration/test/test_power_governor.py]: I see you've removed all the write_log calls.  This data was useful to track power stats from these tests over time.  Is this being captured somewhere else now?

**Review by bgeltz — CHANGES_REQUESTED** [review]

File: integration/test/test_ompt.py (hunk 1 above)
Comment (cmcantalupo) [reply integration/test/test_ompt.py:56]: It is a copy paste error that I found when copy pasting again. This issue may be in other files too...I can check.

File: integration/test/test_power_governor.py (hunk 2 above)
Comment (cmcantalupo) [reply integration/test/test_power_governor.py]: Yes. We had discussed using git to deal with extra files.  

There should also be zero requirement that the tests be run in the git repo now as long as the integration run_config.sh file is sourced.  We can double check that this is true before merging new-profile to dev.

File: integration/test/test_power_governor.py (hunk 3 above)
Comment (cmcantalupo) [reply integration/test/test_power_governor.py]: I'm not sure how to support this feature and also support skip launch.  The log file is created when the launch happens.  These calls to "write_log" are happening in the the assertion part of the code.  We could just write this to standard output?

File: integration/test/test_power_governor.py (hunk 2 above)
Comment (bgeltz) [reply integration/test/test_power_governor.py]: I don't think I was involved in the discussion about using git for those files.

The problem I see there is that we had been using tmp_files to conditionally remove the files in the test if the test passed.  If it failed, the files were saved for debug purposes.  That was the reason that this line exists up in setUpClass() for Python2: ```geopmpy.error.exc_clear()```.

If the idea was to invoke ```git clean``` if a test passed, that will remove all the files from any previously failed tests.  I don't see how git can understand what passed vs. failed.  I'm open to discussion if you've already figured this out though.

File: integration/test/test_power_governor.py (hunk 3 above)
Comment (bgeltz) [reply integration/test/test_power_governor.py]: I'm fine with either a new log file that's specific to this test (so you can just create/open/write to it here, and avoid ```launcher.write_log``` or blasting it all in to stdout.  If it's a separate file, it'll be easier to sift through later.

File: integration/test/test_power_governor.py (hunk 2 above)
Comment (dianarg) [reply integration/test/test_power_governor.py]: One problem that I don't think we ever resolved is that the check in teardown for `sys.exc_info()` is always `(None, None, None)` so it incorrectly cleans up files in python3 even when the test fails.

**Review by bgeltz — APPROVED** [review]

**Pull request merged**